alot/widgets/colour_text.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import re
+from ..settings.const import settings
+
+
+def parse_text_colour(line):
+    """Return colour attribute as (attr, attr_focus)
+
+    :param str line: line of text to be parsed
+    :return: The theme attribute to apply
+    """
+    if settings.get('parse_quotes'):
+        return parse_quotes(line)
+
+
+def parse_quotes(line, max_quote_level=7):
+    """Search for quotes.
+    :param str line: The line of text to be parsed
+    :param int max_quote_level: Search for quotes up to that level
+
+    :return: Theming attribute, None if no quote are available
+    """
+    is_quoted, quote_colour = get_quote_colour(line, max_quote_level)
+
+    if is_quoted and quote_colour is not None:
+        return quote_colour
+    else:
+        return None
+
+
+def get_quote_colour(line, max_quote_level):
+    """Cycle through quotation levels for the line
+
+    :param str line: The line of text to be parsed
+    :param int max_quote_level: Search for quotes up to that level
+    :return: Tuple (is_quoted, quote_colour)
+    """
+    symbol = settings.get('quote_symbol')
+    is_quoted = False
+    quote_colour = None
+
+    for quote_level in range(1, max_quote_level+1):
+        quote_regex = r'^\ *(' + symbol + '\ *){' + str(quote_level) + r'}'
```

Comment:
Why are there backslashes before whitespace? Why are these raw strings? Please use `str.format` instead of `str+str`.

Reply:
You're right, I pushed a commit to use `str.format` instead of `str+str`.

As far as I know however, it is good practice to use raw strings when writing regexes (which may contain special characters). It is not yet the case in this one but I prefer to do it consistently.

I escape the space characters to make them more visible and improve the regex readability. Should I delete them? They are not otherwise necessary.

Reply:
I think escaping stuff that doesn't need escaping does not make the code more readable. It just makes me wonder weather I or the author did not understand (the) regex(es) .

Reply:
Fair enough. I fixed it in the last push.